OpenRA.Game/Graphics/MirrorTileRenderable.cs

```diff
@@ -0,0 +1,57 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Primitives;
+
+namespace OpenRA.Graphics
+{
+	public class MirrorTileRenderable : IRenderable, IFinalizedRenderable
+	{
+		readonly CPos pos;
+		readonly Color color;
+
+		public MirrorTileRenderable(CPos pos, Color color)
+		{
+			this.pos = pos;
+			this.color = color;
+		}
+
+		public WPos Pos => WPos.Zero;
+		public int ZOffset => 0;
+		public bool IsDecoration => true;
+
+		public IRenderable WithZOffset(int newOffset) { return this; }
+
+		public IRenderable OffsetBy(in WVec vec)
+		{
+			return new MirrorTileRenderable(pos, color);
+		}
+
+		public IRenderable AsDecoration() { return this; }
+
+		public IFinalizedRenderable PrepareRender(WorldRenderer wr) { return this; }
+		public void Render(WorldRenderer wr)
+		{
+			var map = wr.World.Map;
+			var uv = pos.ToMPos(map);
+			var r = map.Grid.Ramps[map.Ramp[uv]];
+			var wpos = map.CenterOfCell(uv.ToCPos(map)) - new WVec(0, 0, r.CenterHeightOffset);
+
+			var corners = r.Corners.Select(corner => wr.Viewport.WorldToViewPx(wr.Screen3DPosition(wpos + corner))).ToArray();
```

Comment:
```suggestion
			var corners = r.Corners.Select(corner => wr.Viewport.WorldToViewPx(wr.Screen3DPosition(wpos + corner))).ToList();
```

This isn't the cleanest way about it but it does work. A small perf improvement would be casting to list instead. (List takes up more space in memory but the function is faster, since this is a temporary variable we don't need to save memory)